src/sdk/pynni/nni/batch_tuner/batch_tuner.py

```diff
@@ -88,13 +85,40 @@ def generate_parameters(self, parameter_id):
         ----------
         parameter_id : int
         """
-        self.count +=1
-        if self.count>len(self.values)-1:
+        self.count += 1
+        if self.count > len(self.values) - 1:
             raise nni.NoMoreTrialError('no more parameters now.')
         return self.values[self.count]
 
     def receive_trial_result(self, parameter_id, parameters, value):
         pass
 
     def import_data(self, data):
-        pass
+        """Import additional data for tuning
+        Parameters
+        ----------
+        data:
+            a list of dictionarys, each of which has at least two keys, 'parameter' and 'value'
+        """
+        if len(self.values) == 0:
+            logger.info("Search space has not been initialized, skip this data import")
+            return
+
+        self.values = self.values[(self.count+1):]
+        self.count = -1
+
+        _completed_num = 0
+        for trial_info in data:
+            logger.info("Importing data, current processing progress %s / %s", _completed_num, len(data))
+            # simply validate data format
+            assert "parameter" in trial_info
+            _params = trial_info["parameter"]
+            assert "value" in trial_info
+            _value = trial_info['value']
```

Comment:
local variable `_param` / `_value` will not be actually stored?

Reply:
they are stored in `runhistory`, no need to store them again in this file.

Reply:
but `runhistory` only exists in 'smac_tuner'.

Reply:
yes, so the problem is?

Reply:
I still can't see a data structure in `batch tuner` that store the history data.

Reply:
because batch tuner has already used the imported data, it is reflected on removed configurations from the configuration list/array.